src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -367,20 +384,28 @@ export class NodeClientRequest extends ClientRequest {
 
       this.logger.info('error:\n', error)
 
-      // Suppress certain errors while using the "mock" source.
-      // For example, no need to destroy this request if it connects
-      // to a non-existing hostname but has a mocked response.
-      if (
-        this.responseSource === 'mock' &&
-        NodeClientRequest.suppressErrorCodes.includes(errorCode)
-      ) {
-        // Capture the first emitted error in order to replay
-        // it later if this request won't have any mocked response.
-        if (!this.capturedError) {
-          this.capturedError = error
-          this.logger.info('captured the first error:', this.capturedError)
+      // Suppress only specific Node.js connection errors.
+      if (NodeClientRequest.suppressErrorCodes.includes(errorCode)) {
```

Comment:
This is the biggest change I've added. The previous condition was a bit misleading, so I rewrote it to correctly reflect when the connection errors must be suppressed. 

1. We are only interested in connection errors (those listed in the static `suppressErrorCodes` field on the class. I start the condition from that.
2. Until we've executed all the request listeners, we must suppress all the connection errors and capture the first one to replay it later just in case the listeners won't yield any mocked response and the request will be performed as-is.
3. If we know we are in the mocked response phase, simply ignore all the _connection_ errors (taking advantage of wrapping all these conditions in a higher-level condition). Those aren't relevant. 

The existing test suite has shown that connection errors can occur even in the mocked response phase. I think that's because mocks are applied generally faster than the original responses, and we have to account for that.